### ISSUE TITLE
Dockerized rust applications not receiving signals from command line

### ISSUE THREAD

**Issue by Ch4s3r** [paketo-community/rust#155]
I packaged a rocket server with this rust buildpack but then encountered a problem of not being able to kill it via command line.
This problem is also true for a rocket server with a standard `Dockerfile`.
But adding the `tini` command inbetween solves the signal handling problem here.
The [tini buildpack](https://github.com/paketo-buildpacks/tini) could be added to improve the signal handling (eg. CTRL+C) for dockerized servers in rust.

**Comment by edude03** [comment]
I like the opt out idea because it'll be surprising to people who aren't familiar with docker, but are looking for an easy way to build their rust apps.

**Comment by Ch4s3r** [comment]
@dmikusa-pivotal Rocket is usually handling signals, but just not in containers and I don't know why.
I read this somewhere that this should be like this for most containerized rust apps?

Also regarding opt out should be fine, as this behavior is what i expect from an app, to handle signals properly.

Would always add it, as there is no negative consequence by adding this buildpack from my perspective.

**Comment by dmikusa-pivotal** [comment]
OK, I've implemented this with https://github.com/paketo-community/cargo/pull/146. Take a look at the description there. I'll try to get a release out this week, so you all can give this a try.

My initial tests were working well. The process types generated by Cargo are automatically set to use `tini` (and `tini` is installed). It should be transparent for most use cases. If you see problems by enabling this, definitely let me know.

Also, I added the opt-out. If you just don't want it, like if your app handles signals.
